pkgs/development/python-modules/pythreejs/default.nix

```diff
@@ -0,0 +1,67 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+
+  jupyterlab,
+  setuptools,
+
+  ipywidgets,
+  ipydatawidgets,
+  numpy,
+  traitlets,
+}:
+
+buildPythonPackage rec {
+  pname = "pythreejs";
+  version = "2.4.1";
+  pyproject = true;
+
+  # github sources need to invoke npm, but no package-lock.json present:
+  # https://github.com/jupyter-widgets/pythreejs/issues/419
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-C6UGOnYxLEX2XDds5F9a8zGZSEtM56WeeRrYxNdGqGc=";
+  };
+
+  patches = [
+    # make regex python 3.11 compatible
+    ./fix-re.error.patch
+  ];
```

Comment:
this could probably just be moved to substituteInPlace?

Reply:
The change is a bit annoying to escape, and a patch is a bit more robust.